src/RequestExecutor/HttpRequestExecutor.spec.ts

```diff
@@ -0,0 +1,172 @@
+import nock from 'nock';
+import {
+  anyString,
+  anything,
+  instance,
+  mock,
+  reset,
+  verify,
+  when
+} from 'ts-mockito';
+import { container, Lifecycle } from 'tsyringe';
+
+should();
+
+describe('HttpRequestExecutor', () => {
+  const VirtualScriptsMock = mock<VirtualScripts>(DefaultVirtualScripts);
+  const VirtualScriptMock = mock<VirtualScript>(VirtualScript);
+  const RequestExecutorOptionsMock = mock<RequestExecutorOptions>();
+  const RequestMock = mock<Request>(Request);
+
+  beforeEach(() => {
+    when(VirtualScriptsMock.find(anything())).thenReturn(undefined);
+
+    // default options
+    when(RequestExecutorOptionsMock.timeout).thenReturn(undefined);
+    when(RequestExecutorOptionsMock.proxyUrl).thenReturn(undefined);
+    when(RequestExecutorOptionsMock.headers).thenReturn(undefined);
+    when(RequestExecutorOptionsMock.certs).thenReturn(undefined);
+    when(RequestExecutorOptionsMock.whitelistMimes).thenReturn(undefined);
+    when(RequestExecutorOptionsMock.maxContentLength).thenReturn(undefined);
+    when(RequestExecutorOptionsMock.reuseConnection).thenReturn(undefined);
+
+    container
+      .register<VirtualScripts>(VirtualScripts, {
+        useFactory: () => instance(VirtualScriptsMock)
+      })
+      .register(RequestExecutorOptions, {
+        useFactory: () => instance(RequestExecutorOptionsMock)
+      })
+      .register(
+        RequestExecutor,
+        { useClass: HttpRequestExecutor },
+        { lifecycle: Lifecycle.Singleton }
+      );
+  });
+
+  afterEach(() => {
+    container.reset();
+
+    reset(VirtualScriptsMock);
+    reset(VirtualScriptMock);
+    reset(RequestExecutorOptionsMock);
+    reset(RequestMock);
+  });
+
+  describe('get protocol', () => {
+    it('should return HTTP', () => {
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      expect(executor.protocol).to.eq(Protocol.HTTP);
+    });
+  });
+
+  describe('execute', () => {
+    it('should call setHeaders on the provided request if additional headers were configured globally', async () => {
+      const testHeaders = { testHeader: 'test-header-value' };
+      when(RequestExecutorOptionsMock.headers).thenReturn(testHeaders);
+
+      when(RequestMock.setHeaders(anything())).thenReturn(undefined);
+      const request = instance(RequestMock);
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      await executor.execute(request);
+
+      verify(RequestMock.setHeaders(testHeaders)).once();
+    });
+
+    it('should not call setHeaders on the provided request if there were no additional headers configured', async () => {
+      const request = instance(RequestMock);
+      const executor = container.resolve<RequestExecutor>(RequestExecutor);
+
+      await executor.execute(request);
+
+      verify(RequestMock.setHeaders(anything())).never();
+    });
+
+    it('should transform the request if there is a suitable vm', async () => {
```

Comment:
Let's add one more test to if there is no a suitable vm

Reply:
added, please re-evaluate